pyproject.toml

```diff
@@ -1,5 +1,14 @@
 # Copyright 2022 Canonical Ltd.
 # See LICENSE file for licensing details.
+[project]
+name = "traefik-k8s"
+version = "0.1"  # this is in fact irrelevant
+
+[project.optional-dependencies]
+lib_pydeps = [
+    "opentelemetry-exporter-otlp-proto-grpc",
+    "pydantic<2.0"
+]
```

Comment:
Our deps are already split into reqs.txt, tox.ini, charmcraft.yaml, pydeps.
Do we need this one as well?

Reply:
so I have 3+ tox envs that need to install the pydeps for all of the charm libs we have.
I'd rather have the pydeps-inherited-dependencies all in one place, and I'm not sure how else to do that without repeating myself in all tox envs that need them (and consequently being bothered keeping them in sync)

Reply:
would you rather repeat them in all tox envs? what if we add one more charm lib with its pydeps?

Reply:
Sounds like these need to go into `requirements.txt` in the first place, no?

Reply:
I thought the whole point of pydeps was to be able to avoid that

Reply:
Yeah, we don't need it at pack time because charmcraft pulls it from pydeps, but if we already need it for the test envs, then I think we should:
1. Add them to requriements.txt with a comment `# the deps below are from pydeps and needed here to tox test envs`; or
2. Move _all_ tox deps into one place instead of splitting into yet another place.

Reply:
![image](https://github.com/canonical/traefik-k8s-operator/assets/6230162/3d8dfbfd-44e5-434b-a8df-8b90a5b23714)
<cries in python>